pkg/broker/filter/filter_handler.go

```diff
@@ -274,6 +274,7 @@ func (h *Handler) writeResponse(ctx context.Context, writer http.ResponseWriter,
 			writer.WriteHeader(http.StatusBadGateway)
 			return http.StatusBadGateway, errors.New("received a non-empty response not recognized as CloudEvent. The response MUST be either empty or a valid CloudEvent")
 		}
+		proxyHeaders(resp.Header, writer) // Proxy original Response Headers for downstream use
```

Comment:
I was wondering if we call this function once at line 262, so that we cover every error or success path.

Reply:
Yeah, I can certainly do that.  I did consider it, and the reason I only changed the "success" paths is that I wasn't sure whether it made sense to return certain headers in cases where we are changing the response code.  I still have "Retry-After" on my brain, and that header only "_relates_" to 429/503... so would it make sense to return with 500 or 502.  I suppose in that example the header might not make sense but wouldn't cause a problem?

I will rework the PR (tests mostly) as suggested ; )

Reply:
I see, thanks for the clarification!

/lgtm
/approve